simulation/types/txbuilder.go

```diff
@@ -57,3 +62,62 @@ func (sim *SimCtx) deliverTx(tx sdk.Tx, msg sdk.Msg, msgName string) (simulation
 
 	return simulation.NewOperationMsg(msg, true, "", nil), nil, nil
 }
+
+// GenTx generates a signed mock transaction.
+// TODO: Surely theres proper API's in the SDK for this?
+// (This was copied from SDK simapp, and deleted the egregiously non-deterministic memo handling)
+func genTx(gen client.TxConfig, msgs []sdk.Msg, feeAmt sdk.Coins, gas uint64, chainID string, accNums, accSeqs []uint64, priv ...cryptotypes.PrivKey) (sdk.Tx, error) {
+	sigs := make([]signing.SignatureV2, len(priv))
+	memo := "sample_memo"
+	signMode := gen.SignModeHandler().DefaultMode()
+
+	// 1st round: set SignatureV2 with empty signatures, to set correct
+	// signer infos.
+	for i, p := range priv {
+		sigs[i] = signing.SignatureV2{
+			PubKey: p.PubKey(),
+			Data: &signing.SingleSignatureData{
+				SignMode: signMode,
+			},
+			Sequence: accSeqs[i],
+		}
+	}
+
+	tx := gen.NewTxBuilder()
+	err := tx.SetMsgs(msgs...)
+	if err != nil {
+		return nil, err
+	}
+	err = tx.SetSignatures(sigs...)
+	if err != nil {
+		return nil, err
+	}
+	tx.SetMemo(memo)
+	tx.SetFeeAmount(feeAmt)
+	tx.SetGasLimit(gas)
+
+	// 2nd round: once all signer infos are set, every signer can sign.
+	for i, p := range priv {
+		signerData := authsign.SignerData{
+			ChainID:       chainID,
+			AccountNumber: accNums[i],
+			Sequence:      accSeqs[i],
+		}
+		signBytes, err := gen.SignModeHandler().GetSignBytes(signMode, signerData, tx.GetTx())
```

Comment:
This'll probably reduce logic from [106-115]

```
sigs = []signing.SignatureV2{}
sigs, err := tx.SignWithPrivKey(
			signMode, signerData,
			tx.GetTx, p, gen, accSeqs[i])
if err != nil {
	return nil, err
}
sigsV2 = append(sigs, sigs)
```

Reply:
oh amazing, tysm!